modules/nwtc-library/src/NWTC_IO.f90

```diff
@@ -1475,94 +1475,105 @@ SUBROUTINE AllR16Ary5 (  Ary, AryDim1, AryDim2, AryDim3, AryDim4, AryDim5, Descr
+!! InputFile is the first non-flag argument from the command line. As
+!!     an input to this subroutine, this is used to display the expected
+!!     input file in the help prompt. As an output, it contains the
+!!     name of the input file that is parsed from the command line
+!! ErrStat is an optional argument for catching errors; if present,
+!!     program does not abort on error
+!! Arg2 is the argument given after a flag
+!! Flag is the contents of the first flag argument without the
+!!     leading switch character
    SUBROUTINE CheckArgs ( InputFile, ErrStat, Arg2, Flag )
 
-
       ! Argument declarations:
-   INTEGER,      INTENT(  OUT),OPTIONAL :: ErrStat                                      !< Error status; if present, program does not abort on error
-
-   CHARACTER(*), INTENT(INOUT)          :: InputFile                                    !< The name of the input file specified on the command line. (first non-flag argument)
-   CHARACTER(*), INTENT(  OUT),OPTIONAL :: Arg2                                         !< an optional 2nd non-flag argument from command line
-   CHARACTER(*), INTENT(  OUT),OPTIONAL :: Flag                                         !< an optional flag argument (e.g., restart); the last argument starting with a switch character. 
+   CHARACTER(*), INTENT(INOUT)           :: InputFile          !< The name of the input file specified on the command line.
+   INTEGER,      INTENT(  OUT), OPTIONAL :: ErrStat            !< An optional argument for catching errors; if present, program does not abort on error.
+   CHARACTER(*), INTENT(  OUT), OPTIONAL :: Arg2               !< An optional 2nd non-flag argument from command line.
+   CHARACTER(*), INTENT(  OUT), OPTIONAL :: Flag               !< An optional flag argument; the first argument starting with a switch character. 
 
       ! Local declarations:
-
-   INTEGER                              :: IArg                                         ! The argument number.
-   INTEGER                              :: NumArg                                       ! The number of arguments on the command line.
-                                        
-   INTEGER                              :: Error                                        ! Error Status: indicates if there was an error getting an argument.
-   LOGICAL                              :: FirstArg                                     ! flag to determine if it's the first non-switch argument
-   CHARACTER(LEN(InputFile))            :: Arg                                          ! A command-line argument.
-   
+   INTEGER                               :: NumArg             ! The number of arguments on the command line.
+   CHARACTER(LEN(InputFile))             :: Arg                ! A command-line argument.
 
       ! Find out how many arguments were entered on the command line.
-   NumArg   = COMMAND_ARGUMENT_COUNT()
-   FirstArg = .TRUE.
+   NumArg = COMMAND_ARGUMENT_COUNT()
 
-   IF ( PRESENT(Arg2) ) Arg2 = ""
```

Comment:
I think you should leave these lines in so that these (optional) arguments always get initialized, regardless of how many command line arguments exist.

Reply:
Good point - done